pkg/helmvalues/helmvalues.go

```diff
@@ -106,3 +106,32 @@ func (v *HelmValues) AddListValue(location string, newValue string) error {
 func (v *HelmValues) AddValues(newValues *HelmValues) error {
 	return deepmerge.DeepMerge(v.values, newValues.values, deepmerge.Config{})
 }
+
+// ClearFeatureGates removes all feature gate settings from the helm values
+// This is used to ensure cluster feature gate configuration takes precedence
+// over any feature gates defined in helm values files
+func (v *HelmValues) ClearFeatureGates() error {
+	// Clear operator-controller feature gates if they exist
+	if _, found, _ := unstructured.NestedStringSlice(v.values, strings.Split(EnableOperatorController, ".")...); found {
```

Comment:
Retrieving the `NestedStringSlice()` is not necessary. Everyone of these can simply be a `SetNestedStringSlice()`.